store/models/orm.go

```diff
@@ -228,6 +229,32 @@ func (orm *ORM) BridgeTypeFor(name string) (BridgeType, error) {
 	return tt, err
 }
 
+// MarkRan will set Ran to true for a given initiator
+func (orm *ORM) MarkRan(i *Initiator) error {
+	dbtx, err := orm.Begin(true)
+	if err != nil {
+		return err
+	}
+	defer dbtx.Rollback()
+
+	var ir Initiator
+	if err := orm.One("ID", i.ID, &ir); err != nil {
+		return err
+	}
+
+	if !ir.Ran {
+		ir.Ran = true
+	} else {
+		return fmt.Errorf("Job runner: Initiator: %v cannot run more than once", ir.ID)
+	}
+
+	ir.Ran = true
```

Comment:
You already set this to true on line 246, so you can remove this line.

Reply:
Style wise, I kind of prefer getting rid of the top version, and just having the one error check: `if ir.Ran`. `if !... else` can always be flipped to avoid the negation, and having only the error indented follows the suggested Go style of [indent error cases](https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow), and leave the happy path outdented.

Reply:
I actually already did that :grinning: